### DISCUSSION TITLE
欢迎加入讨论~

### DISCUSSION THREAD

**Discussion by Wcowin** [Wcowin/Wcowin.github.io#7] [Show and tell]
可以交流经验，共同学习

**Comment by arv-anshul** [comment]
Hey @Wcowin , I am from [`mkdocs-material` repo's discussions](https://github.com/squidfunk/mkdocs-material/discussions/6974).

I like to express my thoughts on your personal website. I've looked into it and I feel it is huge, really huge if you compared to mine [website](https://arv-anshul.github.io).

### The Thoughts

- You can do some improvement/changes/modification in your website in order to make it more organised and easy to navigate for others because while exploring your website I have no clue what am I going to see in next page 🤔.
- You can categorise your notes/journals/thoughts/blogs more rigorously; for example: you can use [`blog`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/) plugin to maintan you blogs.
- For your college notes, you can create new `tag` for that or create a new **Nav Item/Section**.

These are just my thoughts you can ignore this if you want. Thanks!

**Reply by Wcowin** [reply]
Thank you Anshul Raj Verma, I see you are from India, it is a beautiful country! I'm glad you are so interested in my blog, I also visited your website and was pleasantly surprised by your work, your website is so simple and nice to look at. Thank you for your suggestions. I will listen to your suggestions and improve them. I have been sorting out the Bolg part recently and noticed the confusion of content classification. I will improve these issues.
Indian friend Anshul Raj Verma, I am considering adding you to my friend chain. I hope you can give me information in the following format so that I can add it:
***
>Name: Wcowin's Web
>Link: https://wcowin.work/
>Avatar: https://s2.loli.net/2024/02/01/gaE47y5fKM6kosV.png
>Introduction: Follow this arduous journey to reach the stars

I also have a question, how is your blog section optimized? I think it looks great! Several people from China and I are making MKdocs themes for China.

**Reply by arv-anshul** [reply]
Thanks @Wcowin , for your kind reply.

> I hope you can give me information in the following format so that I can add it:

> [!note]
>
> **Name:** Anshul Raj Verma
> **Link:** arv-anshul.github.io/about
> **Avatar:** https://avatars.githubusercontent.com/u/111767754?v=4
> **Introduction:** Let's learn together and Build together.

**Reply by arv-anshul** [reply]
> I also have a question, how is your blog section optimized?

There is no additional optimisation by me, it is what comes in the box already. I have just customize the look which you already know from [`mkdocs-material` repo's disscussion](https://github.com/squidfunk/mkdocs-material/discussions/6974).

**Reply by arv-anshul** [reply]
> I think it looks great! Several people from China and I are making MKdocs themes for China.

That's great. But can I know, how and what are you specifically making. Is there any additional information for that or any kind of proof of work because I don't get you that "you and your friends are making `mkdocs` theme for China".

Sorry, for the confusion 😞.

**Reply by Wcowin** [reply]
Sorry I may have misrepresented myself, we are just doing a MKdocs blog optimized for the China region (not an official act) just a hobby , in line with our usage habits!
I've learned a lot from your blog, the work you do is amazing and you are a great programmer.
I followed you and added your [link to my blog](https://wcowin.work/about/link.html)~!
Thanks again for your serious reply!
<img width="428" alt="image" src="https://github.com/Wcowin/Wcowin.github.io/assets/99159173/4336908a-96a8-4e07-90c7-18d9fc93192b">

**Reply by Wcowin** [reply]
<img width="936" alt="image" src="https://github.com/Wcowin/Wcowin.github.io/assets/99159173/1f30e8f0-eee0-44ca-b9e7-67c2071d92c8">

May I ask how you did this, I didn't work out the lines of code yesterday

```html
<p align="center" markdown>
{% for social in about.socials|sort(attribute="title") %}
<a href="{{ social.url }}" title="{{ social.title }}" > :{{ social.icon }}:{ .lg .light } </a>&nbsp; &nbsp;
{% endfor %}
</p>
```

**Reply by arv-anshul** [reply]
![about.md](https://private-user-images.githubusercontent.com/99159173/320188944-1f30e8f0-eee0-44ca-b9e7-67c2071d92c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzODk5ODYsIm5iZiI6MTcxMjM4OTY4NiwicGF0aCI6Ii85OTE1OTE3My8zMjAxODg5NDQtMWYzMGU4ZjAtZWVlMC00NGNhLWI5ZTctNjdjMjA3MWQ5MmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA2VDA3NDgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ODkwYjMzODhhYTcwNTcwOTBiZjI3Y2ZlNTY0ZTZlY2UyZTI1NzlhNGJlODAxNDFhMjgwOGZjMmUzOTUxN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Fo3o2v_vERtb69PZ4X5STENZv6z9xYIMuECp_iPnYPs)

I have defined a [`yaml`](https://github.com/arv-anshul/arv-anshul.github.io/tree/main/docs/data/render_yaml/about.yaml) file where I defined all my social links which gets render here using that code-block (which you quoted in the above comment/reply).

For, rendering I am using a plugin [`mkdocs-markdownextradata-plugin`](https://pypi.org/project/mkdocs-markdownextradata-plugin/), for better understanding about this plugin please refer to its documentation.

**Reply by w1ndys** [reply]
Thank you for your discussion. I have learned something new here. I am going to use yaml to build my template breeding.
@Wcowin @arv-anshul

**Reply by Wcowin** [reply]
> ![about.md](https://private-user-images.githubusercontent.com/99159173/320188944-1f30e8f0-eee0-44ca-b9e7-67c2071d92c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzODk5ODYsIm5iZiI6MTcxMjM4OTY4NiwicGF0aCI6Ii85OTE1OTE3My8zMjAxODg5NDQtMWYzMGU4ZjAtZWVlMC00NGNhLWI5ZTctNjdjMjA3MWQ5MmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA2VDA3NDgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ODkwYjMzODhhYTcwNTcwOTBiZjI3Y2ZlNTY0ZTZlY2UyZTI1NzlhNGJlODAxNDFhMjgwOGZjMmUzOTUxN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Fo3o2v_vERtb69PZ4X5STENZv6z9xYIMuECp_iPnYPs)
> 
> 
> 
> I have defined a [`yaml`](https://github.com/arv-anshul/arv-anshul.github.io/tree/main/docs/data/render_yaml/about.yaml) file where I defined all my social links which gets render here using that code-block (which you quoted in the above comment/reply).
> 
> 
> 
> For, rendering I am using a plugin [`mkdocs-markdownextradata-plugin`](https://pypi.org/project/mkdocs-markdownextradata-plugin/), for better understanding about this plugin please refer to its documentation.

I understand this work, I will give it a try, keep watching! Thanks for the support.

**Reply by Wcowin** [reply]
> Thank you for your discussion. I have learned something new here. I am going to use yaml to build my template breeding.
> 
> @Wcowin @arv-anshul 

@arv-anshul Do this work I think is very good, recommend using
[圆角化设计](https://github.com/squidfunk/mkdocs-material/discussions/6974)

**Comment by arv-anshul** [comment]
- I'm thinking about maintaining similar pages like you, for example: recommendations page (movies, shows, Dev tools, Browser Extensions, etc.), programming topics references page and more.
- Also thinking about maintaining a page where I put my friend's infos like as you do.

All thanks to you @Wcowin for these inspirations.

**Comment by w1ndys** [comment]
> ![about.md](https://private-user-images.githubusercontent.com/99159173/320188944-1f30e8f0-eee0-44ca-b9e7-67c2071d92c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzODk5ODYsIm5iZiI6MTcxMjM4OTY4NiwicGF0aCI6Ii85OTE1OTE3My8zMjAxODg5NDQtMWYzMGU4ZjAtZWVlMC00NGNhLWI5ZTctNjdjMjA3MWQ5MmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA2VDA3NDgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ODkwYjMzODhhYTcwNTcwOTBiZjI3Y2ZlNTY0ZTZlY2UyZTI1NzlhNGJlODAxNDFhMjgwOGZjMmUzOTUxN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Fo3o2v_vERtb69PZ4X5STENZv6z9xYIMuECp_iPnYPs)
> 
> I have defined a [`yaml`](https://github.com/arv-anshul/arv-anshul.github.io/tree/main/docs/data/render_yaml/about.yaml) file where I defined all my social links which gets render here using that code-block (which you quoted in the above comment/reply).
> 
> For, rendering I am using a plugin [`mkdocs-markdownextradata-plugin`](https://pypi.org/project/mkdocs-markdownextradata-plugin/), for better understanding about this plugin please refer to its documentation.

Great. I succeeded in recreating him.

https://ezqfnu.w1ndys.top/test/#yaml

**Comment by arv-anshul** [comment]
Hi @Wcowin once again, I just want to ask that do you use any **formatter tool** to format and manage your docs.

Recently, I have discovered [`mdformat-mkdocs`](https://pypi.org/p/mdformat_mkdocs) but it has some limitations that's why I raised [a issue](https://github.com/KyleKing/mdformat-mkdocs/issues/45).

Just want to know your POV on this. Thanks.
